flows/waits/time.go

```diff
@@ -0,0 +1,25 @@
+package waits
+
+import (
+	"github.com/nyaruka/goflow/flows"
+	"github.com/nyaruka/goflow/flows/events"
+)
+
+const TypeTime string = "time"
+
+type TimeWait struct {
+	Timeout int `json:"timeout" validation:"required,gte=0"`
+}
+
+func (w *TimeWait) Type() string { return TypeMsg }
+
+func (w *TimeWait) Apply(run flows.FlowRun, step flows.Step) {
+	run.ApplyEvent(step, nil, &events.TimeWaitEvent{Timeout: w.Timeout})
+}
+
+// CanResume returns true for a message wait if a message has now been received on this step
+func (w *TimeWait) CanResume(run flows.FlowRun, step flows.Step) bool {
+	return true
```

Comment:
The caller should either be waiting from the creation time of the wait or we should use absolute numbers. The whole name`CanResume` implies the wait is making the decision here, it makes no sense to have it do nothing.

Does `msg_wait` also do nothing and just assume there is a `msg_received`?